### PR TITLE
Hide package metadata for certain packages

### PR DESCRIPTION
**Pull request by shishirx34**
The package details will be hidden for certain packages which are unlisted, locked and if any of the owner is locked.

Spec: https://github.com/NuGet/Engineering/pull/4257

### CODE REVIEW

**Review by advay26 — DISMISSED** [review]

File: src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs [hunk 1]
```diff
@@ -156,6 +156,13 @@ public bool BlockSearchEngineIndexing
             }
         }
 
+        public bool ShouldHideDetailsAndLinks
```
Comment (agr) [inline src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:159]: Negative properties are not very readable: all its uses will be prepended with `!` operator creating double negation, which can be confusing. How about we have `ShowDetailsAndLinks` property instead?

File: src/NuGetGallery/ViewModels/ReportAbuseViewModel.cs [hunk 2]
```diff
@@ -27,5 +27,13 @@ public class ReportAbuseViewModel : ReportViewModel
         [StringLength(4000)]
         [Display(Name = "Details")]
         public string Message { get; set; }
+
+        public bool ShouldHideReportAbuseForm
```
Comment (agr) [inline src/NuGetGallery/ViewModels/ReportAbuseViewModel.cs:31]: Same sentiment as above. It should be `ShowReportAbuseForm`.

File: src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs (hunk 1 above)
Comment (agr) [reply src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:159]: I would also avoid using `Should` in general: it's just a visual noise. If you decide to keep `Hide`, please at least rename it to just `HideDetailsAndLinks`.

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml [hunk 3]
```diff
@@ -15,138 +15,150 @@
...
                 )
             }
-            <p tabindex="0">
-                <text>
-                    Please provide a detailed abuse report with evidence to support your claim! We cannot delete packages without evidence that they exhibit malicious behavior.
-                </text>
-            </p>
-
-            @using (Html.BeginForm())
+             else
```
Comment (agr) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml:26]: indentation?

**Review by agr — DISMISSED** [review]

File: src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs (hunk 1 above)
Comment (shishirx34) [reply src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:159]: I think `HideDetailsAndLinks` shows the correct intention, also avoids the usage of not operator in the cshtml.

Comment (shishirx34) [reply src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:159]: nevermind.. agree double negation is confusing. First suggestion seems better.

File: src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs [hunk 4]
```diff
@@ -156,10 +156,10 @@ public bool BlockSearchEngineIndexing
             }
         }
 
-        public bool ShouldHideDetailsAndLinks
+        public bool ShowDetailsAndLinks
         {
             get { 
-                return !Listed && Locked && Owners.Any(x => x.IsLocked); 
+                return Listed && !Locked && !Owners.Any(x => x.IsLocked);
```
Comment (agr) [inline src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:162]: Proper inversion of `!Listed && Locked && Owners.Any(x => x.IsLocked)` would be `Listed || !Locked || !Owners.Any(x => x.IsLocked)`.

Comment (agr) [reply src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:162]: Or just `!(!Listed && Locked && Owners.Any(x => x.IsLocked))`, of course.

File: src/NuGetGallery/ViewModels/ReportAbuseViewModel.cs [hunk 5]
```diff
@@ -28,11 +28,11 @@ public class ReportAbuseViewModel : ReportViewModel
         [Display(Name = "Details")]
         public string Message { get; set; }
 
-        public bool ShouldHideReportAbuseForm
+        public bool ShowReportAbuseForm
         {
             get
             {
-                return !IsPackageListed && IsPackageLocked && IsOwnerLocked;
+                return IsPackageListed && !IsPackageLocked && !IsOwnerLocked;
```
Comment (agr) [inline src/NuGetGallery/ViewModels/ReportAbuseViewModel.cs:35]: Same as above. It should be `IsPackageListed || !IsPackageLocked || !IsOwnerLocked`

File: src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs (hunk 4 above)
Comment (shishirx34) [reply src/NuGetGallery/ViewModels/DisplayPackageViewModel.cs:162]: Ah yes, you are right... double negation is indeed confusing :D

**Review by agr — APPROVED** [review]

**Pull request merged**